src/main/scala/org/anish/taxisnow/consumer/DemandConsumer.scala

```diff
@@ -0,0 +1,31 @@
+package org.anish.taxisnow.consumer
+package stream
+
+import org.apache.spark.sql.SparkSession
+
+// Goal is to read from Kafka topic and write to CSV
+
+class DemandConsumer {
+
+  val spark = SparkSession
+    .builder
+    .appName("DemandConsumerToSpark")
+    .getOrCreate()
+
+  val df = spark.readStream
+    .format("kafka")
+    .option("kafka.bootstrap.servers", "localhost:9092")
+    .option("subscribe", "taxis_demand")
+    .load()
+
+  val query = df.writeStream
+    .format("csv")
+    .option("path", "~/Desktop/kafka-test")
+    .option("checkpointLocation", "~/Desktop/checkpoints")
+
+  // start the streaming
+  while(true) {
+    query.start()
```

Comment:
`query.start()` would submit the query and start processing it.
Having this in an infinite loop, will start this infinitely, which would lead to a huge number of queries being created and eventually take down Spark. Try using something like `query.awaitTermination()`